inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape.cpp

```diff
@@ -53,6 +54,19 @@ bool validateStaticShapes(const ngraph::Function& function) {
     return true;
 }
 
+bool propagateUpperBoundFromExistingDSR(std::shared_ptr<ngraph::Function>& function) {
+    bool function_changed = false;
+    for (const auto& op : function->get_ordered_ops()) {
+        if (const auto dsr = ngraph::as_type_ptr<ngraph::vpu::op::DynamicShapeResolver>(op)) {
+            dsr->setMode(ngraph::vpu::op::DynamicShapeResolverMode::INFER_UPPER_BOUND_SHAPE);
+            dsr->validate_and_infer_types();
+            function_changed = true;
+        }
+    }
```

Comment:
Why this function implementation differs from `switchDSRMode` implementation?

setMode for all DSRs + function's validation vs setMode for each DSR and DSR's validation

Reply:
In the last commit, I removed `switchDSRMode`. It was redundant. We can create DSR in tests explicitly with required mode.
In this function we don't use function's validation to save dynamism in the function. Otherwise, upper-bounds would been propagated and no transformation wouldn't been triggered.

Reply:
Thanks for clarification!